glustercli/cmd/device.go

```diff
@@ -2,25 +2,120 @@ package cmd
 
 import (
 	"fmt"
+	"os"
 
+	"github.com/gluster/glusterd2/plugins/device/api"
+
+	"github.com/olekukonko/tablewriter"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 const (
-	helpDeviceCmd    = "Gluster Devices Management"
-	helpDeviceAddCmd = "add device"
+	helpDeviceCmd     = "Gluster Devices Management"
+	helpDeviceAddCmd  = "Add device"
+	helpDeviceListCmd = "List devices in peer"
+	helpDeviceInfoCmd = "Get device info"
 )
 
 func init() {
 	deviceCmd.AddCommand(deviceAddCmd)
+	deviceCmd.AddCommand(deviceListCmd)
+	deviceCmd.AddCommand(deviceInfoCmd)
 }
 
 var deviceCmd = &cobra.Command{
 	Use:   "device",
 	Short: helpDeviceCmd,
 }
 
+var deviceInfoCmd = &cobra.Command{
+	Use:   "info <peerID> <DEVICE>",
+	Short: helpDeviceInfoCmd,
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		devName := args[1]
+		peerID := args[0]
+
+		deviceInfo, err := client.DeviceList(peerID, devName)
+		if err != nil {
+			if GlobalFlag.Verbose {
+				log.WithError(err).WithFields(log.Fields{
+					"device": devName,
+					"peerid": peerID,
+				}).Error("device info failed")
+			}
+			failure("Device info failed", err, 1)
+		}
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetHeader([]string{"Device", "State", "Available Size", "Extent Size", "Used", "Peer ID"})
+		for _, d := range deviceInfo {
+			table.Append([]string{d.Device, d.State, humanReadable(d.AvailableSize),
+				humanReadable(d.ExtentSize), formatBoolYesNo(d.Used), d.PeerID.String()})
+		}
+		table.Render()
+	},
+}
+var deviceListCmd = &cobra.Command{
+	Use:   "list <PeerID>",
+	Short: helpDeviceListCmd,
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var peerID string
+		if len(args) == 1 {
+			peerID = args[0]
+		}
+
+		if peerID == "" {
+			peers, err := client.Peers()
+			if err != nil {
+				if GlobalFlag.Verbose {
+					log.WithError(err).Error("peer list failed")
+				}
+				failure("Failed to get peer list", err, 1)
+			}
+			for _, peer := range peers {
```

Comment:
@Madhu-1  DeviceListHandler is coded in a way that it can handle requests with no peer ids, so I think instead of iterating through all peers we can let handler to handle this request.

Reply:
@Madhu-1  Didn't get you. Doesn't device info struct have peerID as a field ?

Reply:
@vpandey-RH since we are showing Peer Name in CLI output, this approach looks good to me.